mxcubeweb/core/server/resource_handler.py

```diff
@@ -51,8 +51,9 @@ def assert_pydantic_arguments(func):
             continue
 
         # Raise RuntimerError If it's not a Pydantic model
-        if not issubclass(param_type, BaseModel):
-            msg = f"Argument {param_name} of {func} are not a pydantic model"
+        if not issubclass(param_type, BaseModel | float | int | str):
```

Comment:
Also accepting passing of types float, int and str. There is not need to model such simple types. Numbers are considered safe as they are and strings are sanitized before being passed.